### DISCUSSION TITLE
Ruletimer<x> expressions calculation

### DISCUSSION THREAD

**Discussion by 13** [arendst/Tasmota#20091] [Q&A] [answered]
Is there a possibility to increase/decrease an existing Ruletimer<x>?

Example:
```
Ruletimer1=Ruletimer1+100
```

**Comment by sfromis** [comment, marked as answer]
No.
Still, using rules, you could pick up existing an existing countdown value executing `Backlog RuleTimer` and have a rule to pick up the result, add something to it, and issue the command to set the ruletimer again.
Another approach would be to have the expiry trigger restart the ruletimer for the additional period.

**Reply by 13** [reply]
Thank you.
A small example how I solved it:
```
ON Event#readT1 DO Backlog RuleTimer ENDON
ON T1 DO Var10 %value% ENDON
ON Power1#state DO Backlog event readT1; RuleTimer1 Var10+600 ENDON
ON Rules#Timer=1 DO Backlog Var10 0; Power1 0 ENDON
```

**Reply by sfromis** [reply]
> `RuleTimer1 Var10+600`

This will not work, you cannot make additions that way, and you cannot rely on the `event` command and associated trigger being executed before the `RuleTimer` command.

Not sure what you try to do, but maybe `PulseTime` feature may be relevant.
